service/status-report/main.go

```diff
@@ -215,6 +221,29 @@ func setClusterInformation(ctx context.Context, cl client.Client, status interfa
 
 }
 
+func setStorageQuotaUtilizationRatio(ctx context.Context, cl client.Client, status interfaces.StorageClientStatus) {
+	clusterResourceQuota := &quotav1.ClusterResourceQuota{}
+	clusterResourceQuota.Name = status.GetClientName()
+
+	// No need to check for NotFound because unlimited quota client will not have CRQ resource
+	if err := cl.Get(ctx, client.ObjectKeyFromObject(clusterResourceQuota), clusterResourceQuota); client.IgnoreNotFound(err) != nil {
+		klog.Warningf("Failed to get clusterResourceQuota %q: %v", clusterResourceQuota.Name, err)
+	}
+
+	if clusterResourceQuota.Status.Total.Hard != nil {
+		total := clusterResourceQuota.Status.Total.Hard["requests.storage"]
+		used := clusterResourceQuota.Status.Total.Used["requests.storage"]
+		ratio := used.AsApproximateFloat64() / total.AsApproximateFloat64()
+		if ratio > 1 {
+			status.SetStorageQuotaUtilizationRatio(1)
+		} else {
+			status.SetStorageQuotaUtilizationRatio(float32(ratio))
+		}
```

Comment:
1. What if total is 0, you cannot divide by 0
2. Why an `if ... else ...`? it makes more sense using `math.Min` to get the min between the ratio and 1

Reply:
1. Total will never be zero here because we are creating a resource, and we will always have a value greater than 0. We can't generate a token with a value of 0 because that validation is added as part of the token generation API. I have now included the validation for that edge case(If the client admin created a CRQ resource with a quota on a different resource.).
2. math.Min is not available for float32 I have to do lot of to and from conversion like this
`status.SetStorageQuotaUtilizationRatio(float32(math.Min(float64(ratio), float64(1.0))))`

Reply:
Done.